src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -1058,50 +1076,76 @@ static inline JSC::JSValue constructResultObject(JSC::JSGlobalObject* lexicalGlo
+                }
+                case SQLITE_FLOAT: {
+                    *value = jsNumber(sqlite3_column_double(stmt, i));
+                    break;
+                }
+                // > Note that the SQLITE_TEXT constant was also used in SQLite version
+                // > 2 for a completely different meaning. Software that links against
+                // > both SQLite version 2 and SQLite version 3 should use SQLITE3_TEXT,
+                // > not SQLITE_TEXT.
+                case SQLITE3_TEXT: {
+                    size_t len = sqlite3_column_bytes(stmt, i);
+                    const unsigned char* text = len > 0 ? sqlite3_column_text(stmt, i) : nullptr;
+
+                    if (len > 64) {
+                        *value = JSC::JSValue::decode(Bun__encoding__toStringUTF8(text, len, lexicalGlobalObject));
+                        continue;
+                    }
+
+                    *value = jsString(vm, WTF::String::fromUTF8(text, len));
+                    break;
+                }
+                case SQLITE_BLOB: {
+                    size_t len = sqlite3_column_bytes(stmt, i);
+                    const void* blob = len > 0 ? sqlite3_column_blob(stmt, i) : nullptr;
+                    JSC::JSUint8Array* array = JSC::JSUint8Array::createUninitialized(lexicalGlobalObject, lexicalGlobalObject->m_typedArrayUint8.get(lexicalGlobalObject), len);
 
-        for (int i = 0; i < count; i++) {
+                    if (LIKELY(blob && len))
+                        memcpy(array->vector(), blob, len);
 
-            switch (sqlite3_column_type(stmt, i)) {
-            case SQLITE_INTEGER: {
-                // https://github.com/oven-sh/bun/issues/1536
-                result->putDirectOffset(vm, i, jsNumber(sqlite3_column_int64(stmt, i)));
-                break;
-            }
-            case SQLITE_FLOAT: {
-                result->putDirectOffset(vm, i, jsNumber(sqlite3_column_double(stmt, i)));
-                break;
+                    *value = array;
+                    break;
+                }
+                default: {
+                    *value = jsNull();
+                    break;
+                }
+                }
             }
-            // > Note that the SQLITE_TEXT constant was also used in SQLite version
-            // > 2 for a completely different meaning. Software that links against
-            // > both SQLite version 2 and SQLite version 3 should use SQLITE3_TEXT,
-            // > not SQLITE_TEXT.
-            case SQLITE3_TEXT: {
-                size_t len = sqlite3_column_bytes(stmt, i);
-                const unsigned char* text = len > 0 ? sqlite3_column_text(stmt, i) : nullptr;
+        });
 
-                if (len > 64) {
-                    result->putDirectOffset(vm, i, JSC::JSValue::decode(Bun__encoding__toStringUTF8(text, len, lexicalGlobalObject)));
-                    continue;
-                }
+        result = JSC::constructEmptyObject(vm, structure);
```

Comment:
btw @devongovett this is an example of how to populate an object really fast from native code